src/__tests__/decompositions/nnmf.js

```diff
@@ -0,0 +1,23 @@
+import { toBeDeepCloseTo } from 'jest-matcher-deep-close-to';
+
+import { Matrix, NNMF } from '../..';
+
+expect.extend({ toBeDeepCloseTo });
+
+describe('Non-negative Matrix Factorization', () => {
+  it('factorization', () => {
+    let A = new Matrix([
+      [1, 0, 0, 0, 1],
+      [0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 0],
+      [1, 0, 0, 0, 1]
+    ]);
+
+    let nA = new NNMF(A, 4);
+    // nA.doNnmf(1000);
+    nA.doError();
```

Comment:
instead of calling a function to add the error, I would suggest the make the `error` property a getter that calculates it

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

Reply:
Thank you for the advice